code/modules/religion/sparring/sparring_datum.dm

```diff
@@ -0,0 +1,293 @@
+			flub(thrown_by)
+
+/datum/sparring_match/proc/projectile_interference(datum/participant, obj/projectile/proj)
+	SIGNAL_HANDLER
+	if(proj.firer == chaplain || proj.firer == opponent)
+		//oh, well that's allowed. or maybe it isn't. doesn't matter because firing the gun will trigger a violation, so no additional violation needed
+		return
+	var/mob/living/interfering
+	if(isliving(proj.firer))
+		interfering = proj.firer
+	flub(interfering)
+
+///someone randomly fucking died
+/datum/sparring_match/proc/death_flub(datum/deceased)
+	SIGNAL_HANDLER
+
+	flubbed_match()
+
+///someone randomly fucking deleted
+/datum/sparring_match/proc/deletion_flub(datum/qdeleting)
+	SIGNAL_HANDLER
+
+	flubbed_match()
+
+///someone used a gun
+/datum/sparring_match/proc/gun_violation(datum/offender)
+	SIGNAL_HANDLER
+	violation(offender, "using guns")
+
+///someone used a grenade
+/datum/sparring_match/proc/grenade_violation(datum/offender)
+	SIGNAL_HANDLER
+	violation(offender, "using grenades")
+
+///someone used melee weapons
+/datum/sparring_match/proc/melee_violation(datum/offender)
+	SIGNAL_HANDLER
+	violation(offender, "using melee weapons")
+
+/datum/sparring_match/proc/teleport_violation(datum/offender)
+	SIGNAL_HANDLER
+	if(offender == chaplain)
+		end_match(opponent, chaplain, violation_victory = TRUE)
+	else
+		end_match(chaplain, opponent, violation_victory = TRUE)
+
+///someone tried to leave
+/datum/sparring_match/proc/arena_violation(atom/movable/mover, atom/oldloc, direction)
+	SIGNAL_HANDLER
+
+	var/area/inhabited_area = get_area(mover)
+	if(inhabited_area == arena_condition)
+		return //still in the ring!! :)
+
+	violation(mover, "leaving the arena")
+	var/atom/throw_target = get_edge_target_turf(mover, REVERSE_DIR(direction))
+	mover.throw_at(throw_target, 6, 4)
+
+/datum/sparring_match/proc/violation(mob/living/carbon/human/offender, reason)
+	SIGNAL_HANDLER
+
+	to_chat(offender, span_danger("Violation! No [reason]!"))
+	if(offender == chaplain)
+		chaplain_violations_allowed--
+		if(!chaplain_violations_allowed)
+			end_match(opponent, chaplain, violation_victory = TRUE)
+	else
+		opponent_violations_allowed--
+		if(!opponent_violations_allowed)
+			end_match(chaplain, opponent, violation_victory = TRUE)
+
+/datum/sparring_match/proc/flub(mob/living/interfering)
+
+	if(interfering)
+		switch(rand(1,3))
+			if(1)
+				to_chat(interfering, span_warning("You get a bad feeling... for interfering with [chaplain]'s sparring match..."))
+				interfering.AddComponent(/datum/component/omen, TRUE, null, FALSE)
+			if(2)
+				to_chat(interfering, span_warning("[GLOB.deity] has punished you for interfering with [chaplain]'s sparring match!"))
+				lightningbolt(interfering)
+			if(3)
+				to_chat(interfering, span_warning("[GLOB.deity]'s whispers echo through your mind for interfering with [chaplain]'s sparring match!!"))
+				SEND_SOUND(interfering, sound('sound/hallucinations/behind_you1.ogg'))
+				interfering.add_confusion(15)
```

Comment:
this option is pretty weaksauce compared to the others, imo. like, the three options here are "get cursed", "get struck by lightning", or "maybe walk in a slightly different direction than you intended to for a few seconds"